File: app/src/main/java/com/travlog/android/apps/viewmodels/outputs/SearchLocationViewModelOutputs.java
package com.travlog.android.apps.viewmodels.outputs;

import android.support.annotation.NonNull;

import com.travlog.android.apps.models.Prediction;

import java.util.List;

import io.reactivex.Observable;

public interface SearchLocationViewModelOutputs {

    @NonNull
    Observable<List<Prediction>> swapSuggestions();

    @NonNull
    Observable<Prediction> setResultAndBack();
}

File: app/src/main/java/com/travlog/android/apps/viewmodels/outputs/SearchLocationViewModelOutputs.kt
package com.travlog.android.apps.viewmodels.outputs

import com.travlog.android.apps.models.Prediction

import io.reactivex.Observable

interface SearchLocationViewModelOutputs {

    fun swapSuggestions(): Observable<List<Prediction>>

    fun setResultAndBack(): Observable<Prediction>
}
